app/main.py

```diff
@@ -1,8 +1,30 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+        Person.people[name] = self
+
+
+def create_person_list(people: list[dict]) -> list[Person]:
+    for person_dict in people:
+        name = person_dict["name"]
+        age = person_dict["age"]
+        Person(name, age)
+
+    for person_dict in people:
+        name = person_dict["name"]
+        spouse_name = person_dict.get("wife") or person_dict.get("husband")
+
+        if spouse_name:
+            person = Person.people[name]
+            spouse = Person.people.get(spouse_name)
+            if spouse:
+                if "wife" in person_dict:
+                    person.wife = spouse
+                elif "husband" in person_dict:
+                    person.husband = spouse
+
+    return list(Person.people.values())
```

Comment:
you can save values from the first loop and return them here to not loop through all values again